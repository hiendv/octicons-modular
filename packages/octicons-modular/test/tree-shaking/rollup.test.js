/**
 * @jest-environment node
 */

import path from 'path'
import { rollup } from 'rollup'
import resolve from 'rollup-plugin-node-resolve'

describe('Rollup tree-shaking', () => {
  test('works', async () => {
    const code = await roll(path.resolve(__dirname, 'main.js'))
    expect(code).toMatchSnapshot()
  })
})

const roll = async entry => {
  let code = ''
  const bundle = await rollup({
    input: entry,
    plugins: [ resolve() ]
  })

  const { output } = await bundle.generate({
    output: {
      format: 'es'
    }
  })

  for (const chunkOrAsset of output) {
    if (chunkOrAsset.isAsset) {
      continue
    }

    code += chunkOrAsset.code
  }

  return code
}
