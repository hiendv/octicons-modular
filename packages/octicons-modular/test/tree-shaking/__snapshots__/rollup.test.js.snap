// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rollup tree-shaking works 1`] = `
"function factory(e,h){const{width:t,height:a,path:o}=h,i=(h,t)=>{t.label?h[\\"aria-label\\"]=t.label:h[\\"aria-hidden\\"]=!0, t.class?h.class=\`octicon octicon-\${e} \${t.class}\`:h.class=\`octicon octicon-\${e}\`;let a=0===t.scale?0:parseFloat(t.scale)||1,o=a*parseInt(h.width),i=a*parseInt(h.height);return h.width=Number(o.toFixed(2)), h.height=Number(i.toFixed(2)), h},c=e=>Object.keys(e).map(h=>\`\${h}=\\"\${e[h]}\\"\`).join(\\" \\").trim();return{name:e,data:h,svg(e,h=document){let l=h.createElement(\\"div\\");return l.innerHTML=\`<svg \${(e=>{let h=Object.assign({},{scale:1,label:null,class:null},e),o=i({version:\\"1.1\\",width:t,height:a,viewBox:\`0 0 \${t} \${a}\`},h);return c(o)})(e)}>\${o}</svg>\`, l.firstChild}}}var alert$=factory(\\"alert\\",{keywords:[\\"warning\\",\\"triangle\\",\\"exclamation\\",\\"point\\"],path:'<path fill-rule=\\"evenodd\\" d=\\"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z\\"></path>',width:16,height:16});

/* eslint-disable */
console.log(alert$.svg());
// the zap icon is removed since it's unused
"
`;
